app/controllers/auth/SelfAssessmentAction.scala

```diff
@@ -57,11 +57,13 @@ class SelfAssessmentActionImpl @Inject()(
     atsNinoResponse match {
       case SuccessAtsNino(nino) =>
         for {
-          utr <- citizenDetailsService.getUtr(nino)
+          utr <- citizenDetailsService.getMatchingDetails(nino)
```

Comment:
Should probably rename this, its not really a utr but just a MatchingDetailsResponse